### PR TITLE
 Add non watch store

### PR DESCRIPTION
**Pull request by orangedeng**
**Problem:**
For now, we will disable watch function in HPA store when the cluster
version is lower than 1.12. But like GKE, they don't enable HPA v2beta2
api in version > 1.12 yet.

**Solution:**
Use `apiservice` to get the available api groups in the cluster.
For HPA, we will detect `v2beta2.autoscaling` api service to determine
the watch func of hpa store will serve or not.

Related PR:
https://github.com/rancher/types/pull/857

Related issue:
https://github.com/rancher/rancher/issues/20758

### CODE REVIEW

**Comment by orangedeng** [comment]
Rebased

File: pkg/api/store/apiservice/filter.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	}
+}
+
+func (s *noWatchByAPIServiceStore) Watch(apiContext *types.APIContext, schema *types.Schema, opt *types.QueryOptions) (chan map[string]interface{}, error) {
+	clustername := s.manager.ClusterName(apiContext)
+	parts := strings.Split(s.version, "/")
+	if len(parts) == 1 {
+		parts[0] = parts[0] + "."
+	}
+	//revert the splited strings, [autoscaling,v2beta2] become [v2beta2,autoscaling]
```
Comment (gitlawr) [inline pkg/api/store/apiservice/filter.go]: s/revert/reverse/

File: pkg/api/store/apiservice/filter.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+			Store:   store,
+			manager: cm,
+			version: apiVersion,
+		}
+	}
+}
+
+func (s *noWatchByAPIServiceStore) Watch(apiContext *types.APIContext, schema *types.Schema, opt *types.QueryOptions) (chan map[string]interface{}, error) {
+	clustername := s.manager.ClusterName(apiContext)
+	parts := strings.Split(s.version, "/")
```
Comment (gitlawr) [inline pkg/api/store/apiservice/filter.go]: We can extract the version conversion logic(L31-39) to a separate function and add some unit tests for it.

Comment (gitlawr) [reply pkg/api/store/apiservice/filter.go]: Maybe we can use the apiservice name directly and don't need the conversion

Comment (orangedeng) [reply pkg/api/store/apiservice/filter.go]: Those are the api service items in my clusters.
```
v1.                                    Local                        True        25d
v1.apps                                Local                        True        25d
v1.authentication.k8s.io               Local                        True        25d
v1.authorization.k8s.io                Local                        True        25d
v1.autoscaling                         Local                        True        25d
v1.batch                               Local                        True        25d
v1.crd.projectcalico.org               Local                        True        1h
v1.monitoring.coreos.com               Local                        True        1h
v1.networking.k8s.io                   Local                        True        25d
v1.rbac.authorization.k8s.io           Local                        True        25d
v1.storage.k8s.io                      Local                        True        25d
v1beta1.admissionregistration.k8s.io   Local                        True        25d
v1beta1.apiextensions.k8s.io           Local                        True        25d
v1beta1.apps                           Local                        True        25d
v1beta1.authentication.k8s.io          Local                        True        25d
v1beta1.authorization.k8s.io           Local                        True        25d
v1beta1.batch                          Local                        True        25d
v1beta1.certificates.k8s.io            Local                        True        25d
v1beta1.coordination.k8s.io            Local                        True        25d
v1beta1.events.k8s.io                  Local                        True        25d
v1beta1.extensions                     Local                        True        25d
v1beta1.metrics.k8s.io                 kube-system/metrics-server   True        25d
v1beta1.policy                         Local                        True        25d
v1beta1.rbac.authorization.k8s.io      Local                        True        25d
v1beta1.scheduling.k8s.io              Local                        True        25d
v1beta1.storage.k8s.io                 Local                        True        25d
v1beta2.apps                           Local                        True        25d
v2beta1.autoscaling                    Local                        True        25d
v2beta2.autoscaling                    Local                        True        25d
v3.management.cattle.io                Local                        True        6d
v3.project.cattle.io                   Local                        True        1h
```

Most of them would be fine with api service name. But the one `v1.` is not validated. So I still need to use split

**Comment by gitlawr** [comment]
LGTM

**Comment by dramich** [comment]
lgtm

**Pull request merged**